src/main/java/org/killbill/billing/codegen/languages/KillbillJavaGenerator.java

```diff
@@ -196,46 +207,74 @@ public Map<String, Object> postProcessModels(Map<String, Object> objs) {
     // Transform CodegenOperation obj into ExtendedCodegenOperation to add properties into mustache maps
     @Override
     public Map<String, Object> postProcessOperations(Map<String, Object> objs) {
-
         final Map<String, Object> operationsMap = (Map<String, Object>) super.postProcessOperations(objs).get("operations");
         final List<CodegenOperation> operations = (List<CodegenOperation>) operationsMap.get("operation");
         final List<ExtendedCodegenOperation> extOperations = new ArrayList<>(operations.size());
-
         final List<Map<String, String>> imports = (List<Map<String, String>>) objs.get("imports");
-
-
         for (CodegenOperation op : operations) {
             final ExtendedCodegenOperation ext = new ExtendedCodegenOperation(op);
             extOperations.add(ext);
-            if (ext.isReturnModelRefContainer) {
-                addImportIfRequired(imports, String.format("org.killbill.billing.client.model.%s", ext.returnType));
-            }
-            if (ext.isBodyModelRefContainer) {
-                addImportIfRequired(imports, String.format("org.killbill.billing.client.model.%s", ext.bodyParam.dataType));
-            }
-            if (ext.isStream) {
-                addImportIfRequired(imports, "java.io.InputStream");
-                addImportIfRequired(imports, "java.io.OutputStream");
-                addImportIfRequired(imports, "java.net.http.HttpResponse");
-            }
-            if (ext.isListContainer) {
-                addImportIfRequired(imports, "java.util.List");
-            } else if (ext.isMapContainer) {
-                addImportIfRequired(imports, "java.util.Map");
+            addRequiredImports(ext, imports);
+            convertToExtendedCodegenParam(ext, imports);
+            if (shouldAddDateTimeMethod(op)) {
+                addImportIfRequired(imports, "org.joda.time.DateTime");
```

Comment:
I understand why it's there, but I would have expected another part of the code to call `addImportIfRequired(imports, "org.joda.time.LocalDate");` and yet I don't see it - do you know why this is not symmetrical, i.e how the `LocalDate` import ends up in the generated file?

Reply:
Yes, you are right. I checked but I'm not quite sure, I think it has to do with [this ](https://github.com/killbill/killbill-swagger-coden/blob/a8de9958143f1954f53ff6634594e13f1c43909d/src/main/java/org/killbill/billing/codegen/languages/KillbillJavaGenerator.java#L121)line. If I comment this line out, the `import org.joda.time.LocalDate` statement is not inserted into the code. Having said that, [this ](https://github.com/killbill/killbill-swagger-coden/blob/a8de9958143f1954f53ff6634594e13f1c43909d/src/main/java/org/killbill/billing/codegen/languages/KillbillJavaGenerator.java#L120) line similarly adds an import for `org.joda.time.DateTime` but this does automatically insert the corresponding import statement in the code. So, not quite sure what is going on. 😕

Reply:
Yeah, this is strange. It's fine, I was just curious.